src/main/ruby/truffleruby/core/truffle/exception_operations.rb

```diff
@@ -93,18 +93,25 @@ def self.receiver_string(receiver)
 
     def self.message_and_class(exception, highlight)
       message = StringValue exception.message.to_s
+      klass = Primitive.class(exception)
+      klass_name = klass.to_s
 
-      klass = Primitive.class(exception).to_s
       if Primitive.is_a?(exception, Polyglot::ForeignException) and
           Truffle::Interop.has_meta_object?(exception)
-        klass = "#{klass}: #{Truffle::Interop.meta_qualified_name Truffle::Interop.meta_object(exception)}"
+        klass_name = "#{klass_name}: #{Truffle::Interop.meta_qualified_name Truffle::Interop.meta_object(exception)}"
       end
 
       if highlight
-        highlighted_class = " (\e[1;4m#{klass}\e[m\e[1m)"
+        highlighted_class = if klass == RuntimeError && message.empty?
```

Comment:
There is a [rule](https://github.com/andrykonchin/rubocop-truffleruby/blob/master/lib/rubocop/cop/truffleruby/replace_with_primitive_equal.rb) for `Primitive.equal?` but it handles only `#equal?` method calls. It probably makes sense to have handled also `#==` calls with numeric/symbol literal argument (as far as `#==` and `#equal?` should be equivalent and we don't care if any of them was redefined by user)

Reply:
mmh, it's kind of tricky because `foo == SomeConstant` cannot always become `Primitive.equal?`, that only works if the constant is something that makes sense to compare by identity. So probably just leave this as-is and accept we cannot easily catch those cases.